lib/active_snapshot/models/concerns/snapshots_concern.rb

```diff
@@ -34,7 +34,7 @@ def create_snapshot!(identifier: nil, user: nil, metadata: nil)
 
       SnapshotItem.upsert_all(new_entries.map{|x| x.delete("id"); x }, returning: false)
 
-      snapshot
```

Comment:
Instead of reloading couldnt we just set `item_id` and `item_type` in the `create!` call (possibly with a code comment next to it explaining why if thats helpful)

Reply:
Unfortunately not. It's not the snapshot itself, but it's snapshot_items. 
They are in the build state without any `id`. `upsert_all` inserts them in the database, but we can't get the IDs back as the `returning` options only work for [PostgreSQL](https://apidock.com/rails/v7.0.0/ActiveRecord/Persistence/ClassMethods/upsert_all).

I updated the code to be specific on resetting the `snapshot_items` association.